java/ql/src/Security/CWE/CWE-352/JsonpInjection.ql

```diff
@@ -0,0 +1,68 @@
+/**
+ * @name JSONP Injection
+ * @description User-controlled callback function names that are not verified are vulnerable
+ *              to jsonp injection attacks.
+ * @kind path-problem
+ * @problem.severity error
+ * @precision high
+ * @id java/JSONP-Injection
+ * @tags security
+ *       external/cwe/cwe-352
+ */
+
+import java
+import JsonpInjectionLib
+import JsonpInjectionFilterLib
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.deadcode.WebEntryPoints
+import DataFlow::PathGraph
+
+
+/** If there is a method to verify `token`, `auth`, `referer`, and `origin`, it will not pass. */
+class ServletVerifAuth extends DataFlow::BarrierGuard {
+  ServletVerifAuth() {
+    exists(MethodAccess ma, Node prod, Node succ |
+      ma.getMethod().getName().regexpMatch("(?i).*(token|auth|referer|origin).*") and
+      prod instanceof RemoteFlowSource and
+      succ.asExpr() = ma.getAnArgument() and
+      ma.getMethod().getAParameter().getName().regexpMatch("(?i).*(token|auth|referer|origin).*") and
+      localFlowStep*(prod, succ) and
+      this = ma
+    )
+  }
+
+  override predicate checks(Expr e, boolean branch) {
+    exists(Node node |
+      node instanceof JsonpInjectionSink and
+      e = node.asExpr() and
+      branch = true
+    )
+  }
+}
+
+/** Taint-tracking configuration tracing flow from get method request sources to output jsonp data. */
+class JsonpInjectionConfig extends TaintTracking::Configuration {
+  JsonpInjectionConfig() { this = "JsonpInjectionConfig" }
+
+  override predicate isSource(DataFlow::Node source) { source instanceof GetHttpRequestSource }
```

Comment:
I liked your original approach with `RemoteFlowSource` better. This is accurate, but adds some complexity to the query. Wouldn't it be okay to use `RemoteFlowSource` and additionally check if the enclosing callable is a `GET` method?

Reply:
@tamasvajk I made changes based on your suggestions, please review.